shims/spark310/src/main/scala/com/nvidia/spark/rapids/shims/spark310/ParquetCachedBatchSerializer.scala

```diff
@@ -272,17 +274,24 @@ class ParquetCachedBatchSerializer extends CachedBatchSerializer with Arm {
    * Convert an `RDD[ColumnarBatch]` into an `RDD[CachedBatch]` in preparation for caching the data.
    * This method uses Parquet Writer on the GPU to write the cached batch
    *
-   * @param input the input `RDD` to be converted.
-   * @param schema the schema of the data being stored.
+   * @param input        the input `RDD` to be converted.
+   * @param schema       the schema of the data being stored.
    * @param storageLevel where the data will be stored.
-   * @param conf the config for the query.
+   * @param conf         the config for the query.
    * @return The data converted into a format more suitable for caching.
    */
   override def convertColumnarBatchToCachedBatch(input: RDD[ColumnarBatch],
      schema: Seq[Attribute],
      storageLevel: StorageLevel,
      conf: SQLConf): RDD[CachedBatch] = {
+
+    val parquetSchema = getCatalystSchema(schema, schema)
+
     val rapidsConf = new RapidsConf(conf)
+    val broadcastedHadoopConf = getBroadcastedHadoopConf(conf, parquetSchema)
+
+    val broadcastedConf = SparkSession.active.sparkContext.broadcast(conf)
```

Comment:
If we only use these in the else case can we move them to just the else case, so we don't broadcast data we don't need?

Also do we have to broadcast conf?  I thought spark handled that for you without too much difficulty?

Reply:
I double checked its null. Is there a special way to pass it?

Reply:
I thought it would be serialized out with the classes that have it when they are sent to the workers.

Reply:
correction: its not null but empty. When looking up a const like `SQLConf.LEGACY_PARQUET_REBASE_MODE_IN_WRITE.key` blows up with NPE. I will play around with this a bit more to validate your theory. May be I am missing something